jspwiki-event/src/main/java/org/apache/wiki/event/WikiEventManager.java

```diff
@@ -135,8 +135,8 @@ public final class WikiEventManager {
     /* The Map of client object to WikiEventDelegate. */
     private final Map< Object, WikiEventDelegate > m_delegates = new HashMap<>();
 
-    /* The Vector containing any preloaded WikiEventDelegates. */
-    private final Vector< WikiEventDelegate > m_preloadCache = new Vector<>();
+    /* The List containing any preloaded WikiEventDelegates. */
+    private final List< WikiEventDelegate > m_preloadCache = Collections.synchronizedList( new ArrayList<>() );
```

Comment:
pr-wide comment:

please note that, per `Collections.synchronizedList` javadoc, 
> It is imperative that the user manually synchronize on the returned list when traversing it via Iterator, Spliterator or Stream: [...]

enhanced fors use internally an Iterator, so they also must be synchronized. Would you mind rechecking the access to the new Lists in order to ensure they remain synchronized?

Reply:
hi @juanpablo-santos 
Another approach you can take to avoid explicitly synchronizing on the list during iteration is to use a concurrent collection instead of a synchronized collection.

Instead of using Collections.synchronizedList(), we could use java.util.concurrent.CopyOnWriteArrayList or java.util.concurrent.ConcurrentLinkedQueue, both of which are thread-safe and don't require explicit synchronization during iteration.
WDYT?